programs/perpetuals/tests/tests_suite/basic_interactions.rs

```diff
@@ -0,0 +1,245 @@
+        .0;
+    let eth_mint = program_test
+        .add_mint(None, 9, &keypairs[ROOT_AUTHORITY].pubkey())
+        .0;
+
+    // Deploy the perpetuals program onchain as upgradeable program
+    utils::add_perpetuals_program(&mut program_test, &keypairs[PERPETUALS_UPGRADE_AUTHORITY]).await;
+
+    // Start the client and connect to localnet validator
+    let mut program_test_ctx = program_test.start_with_context().await;
+
+    let upgrade_authority = &keypairs[PERPETUALS_UPGRADE_AUTHORITY];
+
+    let multisig_signers = &[
+        &keypairs[MULTISIG_MEMBER_A],
+        &keypairs[MULTISIG_MEMBER_B],
+        &keypairs[MULTISIG_MEMBER_C],
+    ];
+
+    instructions::test_init(
+        &mut program_test_ctx,
+        upgrade_authority,
+        fixtures::init_params_permissions_full(1),
+        multisig_signers,
+    )
+    .await;
+
+    // Initialize and fund associated token accounts
+    {
+        // Alice, 1k USDC
+        {
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &usdc_mint,
+                &keypairs[USER_ALICE].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                1_000_000_000,
+            )
+            .await;
+        }
+
+        // Martin, 100 USDC, 2 ETH
+        {
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &usdc_mint,
+                &keypairs[USER_MARTIN].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                100_000_000,
+            )
+            .await;
+
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &eth_mint,
+                &keypairs[USER_MARTIN].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                2_000_000_000,
+            )
+            .await;
+        }
+
+        // Paul, 150 USDC
+        {
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &usdc_mint,
+                &keypairs[USER_PAUL].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                150_000_000,
+            )
+            .await;
+
+            utils::initialize_token_account(
+                &mut program_test_ctx,
+                &eth_mint,
+                &keypairs[USER_PAUL].pubkey(),
+            )
+            .await;
+        }
+    }
+
+    let (pool_pda, _, lp_token_mint_pda, _, _) = utils::setup_pool_with_custodies_and_liquidity(
+        &mut program_test_ctx,
+        &keypairs[MULTISIG_MEMBER_A],
+        "FOO",
+        &keypairs[PAYER],
+        multisig_signers,
+        vec![
+            utils::SetupCustodyWithLiquidityParams {
+                setup_custody_params: utils::SetupCustodyParams {
+                    mint: usdc_mint,
+                    decimals: 6,
+                    is_stable: true,
+                    target_ratio: 5_000,
```

Comment:
it would be a bit better if you scaled the number based on the decimals (BPS_DECIMALS etc) cause I imagine they might change if we find out I picked the wrong values. But not a big deal.

Reply:
Yhea just saw that in the default unit tests and we discussed it. We will update thanks for the heads up